runtime/src/accounts_background_service.rs

```diff
@@ -606,12 +609,17 @@ impl AccountsBackgroundService {
                     // slots >= bank.slot()
                     bank.flush_accounts_cache_if_needed();
 
-                    if let Some(snapshot_block_height_result) = snapshot_block_height_option_result
-                    {
+                    if let Some(snapshot_handle_result) = snapshot_handle_result {
                         // Safe, see proof above
-                        if let Ok(snapshot_block_height) = snapshot_block_height_result {
+                        if let Ok((snapshot_block_height, snapshot_storages)) =
+                            snapshot_handle_result
+                        {
                             assert!(last_cleaned_block_height <= snapshot_block_height);
                             last_cleaned_block_height = snapshot_block_height;
+                            // Update the option, so the older one is released, causing the release of
+                            // its reference counts of the appendvecs
+                            // Prefix with _ because of the compiler warning - value assigned never read
```

Comment:
another option, if we feel having an actually used variable with `_*` naming is confusing.

```
    let mut last_snapshot_storages = None;
    ...
        let _ = last_snapshot_storages.insert(snapshot_storages);
```
In rust the `_` variable is immediately dropped. Note that this is not the case for something beginning with _. @brooksprumo might also have an opinion on this.

Reply:
I did see reference count problems with Option.  The VecDeque was working fine.

Reply:
tried adding #[allow(unused_assignments)]
before 
last_snapshot_storages = Some(snapshot_storages);

Got the following error.
```
   --> runtime/src/accounts_background_service.rs:624:29
    |
624 | ...                   #[allow(unused_assignments)]
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #15701 <https://github.com/rust-lang/rust/issues/15701> for more information

Reply:
In my prior testing VecDeque was holding the reference counts well.  Maybe it can be reconsidered?

Reply:
In my prior tests, I saw the reference count problem happen when using the Option (probably due to _ prefex).  But when using VecDeque, the problem didn't happen, probably there is read operation when dequeueing happens.

Reply:
Because we hardlink the appendvec file in <account_path>/run/ into the <account_path>/snapshot/<slot>/, we don't want the file content to be changed (recycling would cause it).  So we want the reference count to be kept to avoid the file content change.  When the validator software process exits, there is no more change for that to happen.

Reply:
Ok, added that debug!.